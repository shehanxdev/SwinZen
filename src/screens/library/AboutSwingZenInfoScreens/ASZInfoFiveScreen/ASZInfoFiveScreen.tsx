import React from 'react';
import { View } from 'react-native';

import { Text } from '@sz/components';
import { tw } from '@sz/config';
import { AboutSZInfoFivePoints, Color, TextAlignment, TextVariant } from '@sz/constants';

import { BaseLibraryScreen } from '../../components';

const TEST_ID_PREFIX = 'ASZInfoFiveScreenTestID';

export function ASZInfoFiveScreen() {
  return (
    <BaseLibraryScreen>
      <View style={tw`mx-4 mt-8`} testID={TEST_ID_PREFIX}>
        <Text color={Color.Neutral.White} variant={TextVariant.Body1SemiBold} textAlign={TextAlignment.Auto}>
          {`Note this is available only with paid subscription.\n`}
        </Text>
        {AboutSZInfoFivePoints.map((data, index) => (
          <View key={`${TEST_ID_PREFIX}-Point-${index}`}>
            <Text color={Color.Neutral.White} variant={TextVariant.Body1SemiBold} textAlign={TextAlignment.Auto}>
              {data.title}
            </Text>
            <Text color={Color.Neutral.White} variant={TextVariant.Body1Regular} textAlign={TextAlignment.Auto}>
              {data.body}
              {`\n`}
            </Text>
          </View>
        ))}
      </View>
    </BaseLibraryScreen>
  );
}
