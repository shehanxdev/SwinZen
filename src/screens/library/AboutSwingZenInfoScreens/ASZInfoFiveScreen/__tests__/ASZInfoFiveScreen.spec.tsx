import { render } from '@testing-library/react-native';
import * as React from 'react';

import { ASZInfoFiveScreen } from '../ASZInfoFiveScreen';

describe('ASZInfoFive Screen', () => {
  const testID = 'ASZInfoFiveScreenTestID';

  const getRenderedScreen = () => render(<ASZInfoFiveScreen />);

  it(`should render ASZInfoFive Screen correctly`, () => {
    const renderer = getRenderedScreen();
    const renderTree = renderer.toJSON();
    expect(renderTree).toMatchSnapshot();
  });

  it('should find the ASZInfoFive Screen via testID', () => {
    const { getByTestId } = getRenderedScreen();
    const foundScreen = getByTestId(testID);
    expect(foundScreen).toBeTruthy();
  });
});
