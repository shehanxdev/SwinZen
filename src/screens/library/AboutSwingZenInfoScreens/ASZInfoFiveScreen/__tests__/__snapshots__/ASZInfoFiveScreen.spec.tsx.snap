// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ASZInfoFive Screen should render ASZInfoFive Screen correctly 1`] = `
<BVLinearGradient
  colors={
    [
      4279917603,
      4278984207,
      4278650631,
    ]
  }
  endPoint={
    {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    {
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "position": "relative",
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      {
        "testUri": "../../../src/assets/images/grassBackground.jpg",
      }
    }
    style={
      {
        "opacity": 0.1,
        "position": "absolute",
      }
    }
  />
  <RCTSafeAreaView
    style={
      {
        "height": "100%",
        "paddingTop": 50,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        [
          {
            "flexBasis": "0%",
            "flexGrow": 1,
            "flexShrink": 1,
          },
          {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <RCTScrollView
        contentContainerStyle={
          {
            "flexGrow": 1,
          }
        }
        keyboardShouldPersistTaps="handled"
      >
        <View>
          <View
            style={
              {
                "marginLeft": 16,
                "marginRight": 16,
                "marginTop": 32,
              }
            }
            testID="ASZInfoFiveScreenTestID"
          >
            <Text
              style={
                [
                  {
                    "color": "#FFFFFF",
                    "lineHeight": 28,
                  },
                  {
                    "fontFamily": "SourceSansPro-SemiBold",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "textAlign": "auto",
                  },
                ]
              }
            >
              Note this is available only with paid subscription.

            </Text>
            <View>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-SemiBold",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Setup/Back Knee Angle:
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 18,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Measures angle of back knee. If too straight or too bent at address.
                

              </Text>
            </View>
            <View>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-SemiBold",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Setup/Wrist Angle:
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 18,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Measures angle of arms and club shaft at address.
                

              </Text>
            </View>
            <View>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-SemiBold",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Setup/Posture:
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 18,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Measures angle of proper posture and spine tilt at address.
                

              </Text>
            </View>
            <View>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-SemiBold",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Backswing/Swing Plane:
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 18,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Determines if the club shaft is on the proper plane at Â½ backswing.
                

              </Text>
            </View>
            <View>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-SemiBold",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Backswing/Stable Back Leg:
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 18,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Tracks the angle of the back leg thru backswing to determine if excessive change.
                

              </Text>
            </View>
            <View>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-SemiBold",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Backswing/Stable Spine:
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 18,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Tracks the spine angle thru backswing to determine if excessive change.
                

              </Text>
            </View>
            <View>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-SemiBold",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Backswing/Shoulder Tilt:
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 18,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Tracks the shoulder tilt thru backswing to determine if shoulders are at proper angle and not too steep.
                

              </Text>
            </View>
            <View>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-SemiBold",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Downswing/Swing Plane:
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 18,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Determines if the club shaft is on the proper plane at Â½ downswing.
                

              </Text>
            </View>
            <View>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-SemiBold",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Downswing/Club Path:
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 18,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Tracks clubhead path thru impact to determine if inside out or outside in.
                

              </Text>
            </View>
            <View>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-SemiBold",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Downswing/Stable Spine:
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 18,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Tracks the spine angle thru downswing to determine if excessive change.
                

              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>
</BVLinearGradient>
`;
